components/external-link/index.js

```diff
@@ -8,33 +8,76 @@ import { compact, uniq } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Dashicon from '../dashicon';
+import Button from '../button';
 import './style.scss';
 
-function ExternalLink( { href, children, className, rel = '', ...additionalProps } ) {
-	rel = uniq( compact( [
-		...rel.split( ' ' ),
-		'external',
-		'noreferrer',
-		'noopener',
-	] ) ).join( ' ' );
-	const classes = classnames( 'components-external-link', className );
-	return (
-		<a { ...additionalProps } className={ classes } href={ href } target="_blank" rel={ rel }>
-			{ children }
+class ExternalLink extends Component {
+	getRel() {
+		const { rel = '' } = this.props;
+
+		// Allow to omit the `rel` attribute passing a `null` value.
```

Comment:
>the rel prop default value is still external noreferrer noopener but it wouldn't be appropriate for internal links so it can be omitted passing a null value

What is the consequence of applying them anyways? Purely in the interest of a simpler implementation.

__Edit:__ Alternatively, is there any way we could programmatically determine whether we'd want to omit the `rel` ? What's the criteria ? That it's within the same hostname ?

Reply:
Yep, I thought about that and also looked if there was some `url` utility to determine when the link is internal or external. I'm afraid that would get a bit complex, there are different cases to cover e.g.:
`options-permalink.php`
`/something`
`http://site-domain/something`
etc.
On the other hand, is the rel value `external` really necessary? I wouldn't mind to simplify, remove it and keep `noreferrer noopener` also for internal links. Thoughts?